### ISSUE TITLE
How to modify extension to only open maximize a concret page

### ISSUE THREAD

**Issue by dimaiam** [ericchase/addon--maximize-all-windows#6]
Hello!!

I love this extension!, thanks for your work!

But, I need to modify it to only open maximize a concret URL or tab in chrome.

It is possible to do it?

I expend to many hours but I don't know how to program in javascript.

thanks!

**Comment by ericchase** [comment]
You want to maximize tabs with a specific URL?

**Comment by dimaiam** [comment]
> You want to maximize tabs with a specific URL?

I work in a Hospital, and there we use the program "SAP", in that program we recently change one important feature that works with Chrome, is like an applicattion, the storage tab of that application is https://etc.argos.intranet.gencat.cat/, the problem with that is like open always soo smallest, like 1024p, and we use your extension to always open up like maximized. And works great.

But recently we have to add a new extension from another company, is "Yeastar Linkus for Google", and the problem is like that application doesn't have to open maximize way.

Is possible to do that?

**Comment by ericchase** [comment]
Yes, this is possible, but this would add extra complexity to my extension which doesn't make sense, because only 1 person needs this complexity.

You really need a new extension altogether.

**Comment by ericchase** [comment]
Perhaps I can make a cousin extension like "auto-resize-target-window" or something.

**Comment by dimaiam** [comment]
> Perhaps I can make a cousin extension like "auto-resize-target-window" or something.

Well, it's clear that I don't want you to make an extension for me, I was just wondering if you could guide me a little to make it possible.

in any case, thanks.

**Comment by ericchase** [comment]
No, it shouldn't be too difficult. And I have a developer account on Google's Chrome store, so I can put it up easily

**Comment by ericchase** [comment]
If you're interested in how it would be done, then I could write the developer process in the readme file

**Comment by dimaiam** [comment]
> If you're interested in how it would be done, then I could write the developer process in the readme file
well, i can only say thank you.  and of course you don't need to explain your code.

**Comment by ericchase** [comment]
No worries! I teach this stuff, so I am happy to if you're interested in how some of it works. I'll try to finish it this week and have it published so you can test it out soon.

**Comment by ericchase** [comment]
Update: I've rewritten the core addon for **manifest v3**.

Now I will be adding an option to add window titles/urls for skipping.

**Comment by dimaiam** [comment]
thanks for the feedback!! sure it will be fantastic

**Comment by ericchase** [comment]
Very sorry. Adding the *skip specific urls* feature is a bit more complicated than what I expected. I've gotten closer, but there's still a couple things I have to figure out.

![image](https://user-images.githubusercontent.com/30473157/171175456-c68d7f2c-2551-4259-8d12-314a9f7cd3c5.png)

**Comment by ericchase** [comment]
So, I took a hiatus from working on this for a while.

Today, I made a beta version of the addon featuring only the things that work right now. Google's manifest v3 update with Chrome changed how a lot of things work on the backend, so I did my best to keep similar features from before.

I wish I could have added the pattern exclusion stuff, but I had to make some drastic changes just to get this watered down version submitted.

I'll let you know if I'm able to make progress with the pattern exclusions later
